tests/accelerators/test_accelerator_connector.py

```diff
@@ -506,6 +506,15 @@ def test_accelerator_cpu(_):
         trainer = Trainer(accelerator="cpu", gpus=1)
 
 
+@mock.patch("torch.cuda.is_available", return_value=False)
```

Comment:
@rohitgr7 

Here, we assumed the accelerator is not available. What if it's available and `devices="0"/0/ []]`?

It leads to such errors
```bash
    self._parallel_devices = self.accelerator.get_parallel_devices(self._devices_flag)
  File "/home/jovyan/pytorch-lightning/pytorch_lightning/accelerators/gpu.py", line 82, in get_parallel_devices
    return [torch.device("cuda", i) for i in devices]
TypeError: 'NoneType' object is not iterable
```

It is addressed in #12633